mqtt-client/src/main/java/com/gojek/mqtt/connection/MqttConnection.kt

```diff
@@ -484,6 +484,16 @@ internal class MqttConnection(
                     timeTakenMillis = (clock.nanoTime() - subscribeStartTime).fromNanosToMillis()
                 )
                 runnableScheduler.scheduleMqttHandleExceptionRunnable(mqttException, true)
+            } catch (illegalArgumentException: IllegalArgumentException) {
+                connectionConfig.connectionEventHandler.onMqttSubscribeFailure(
+                    topics = topicMap,
+                    throwable = illegalArgumentException,
+                    timeTakenMillis = (clock.nanoTime() - subscribeStartTime).fromNanosToMillis()
+                )
+                runnableScheduler.scheduleMqttHandleExceptionRunnable(
```

Comment:
no need to do this. this is responsible for reconnection. we can remove this

Reply:
Sure